src/node_buffer.cc

```diff
@@ -1211,6 +1211,41 @@ void DetachArrayBuffer(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void Btoa(const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 1);
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_IF_NOT_STRING(env, args[0], "argument");
+
+  Local<String> input = args[0].As<String>();
+  MaybeStackBuffer<char> buffer;
+  size_t written;
+
+  if (input->IsExternalOneByte()) {  // 8-bit case
+    auto ext = input->GetExternalOneByteStringResource();
+    size_t expected_length = simdutf::base64_length_from_binary(ext->length());
+    buffer.AllocateSufficientStorage(expected_length + 1);
+    buffer.SetLengthAndZeroTerminate(expected_length);
+    written = simdutf::binary_to_base64(
+        ext->data(), ext->length(), buffer.out(), simdutf::base64_default);
+  } else {  // 16-bit case
```

Comment:
These comments look a bit misleading in that they seem to imply that `input` would always be stored as a 16-bit string if this branch is taken, which is not the case (most of the time it will still be an 8-bit string, just not an external one)

Reply:
Thanks! Updated the comment.

Reply:
@addaleax 

> These comments look a bit misleading in that they seem to imply that input would always be stored as a 16-bit string if this branch is taken, which is not the case (most of the time it will still be an 8-bit string, just not an external one)

In https://github.com/nodejs/node/pull/52428, we call `String::Value value(isolate, str);` which stores the string using 16-bit units. Within v8, the string might be a one-byte string, but we process it as a 16-bit string (that's what `String::Value value(isolate, str);` does).